oeps/oep-0011.rst

```diff
@@ -0,0 +1,299 @@
+   own brand. This includes elements as simple as overriding the fonts,
+   imagery and color schemes, to more complex customization of headers,
+   footers and registration flows.
+
+   You can learn more about how Open edX supports theming with
+   the guide `Changing Themes for an Open edX Site`_. One simple way to
+   make your interface themeable is to use patterns from the
+   `edX Pattern Library`_ as they already support theming.
+
+8. **Front end code should be safe from cross site scripting**
+
+   Cross Site Scripting (XSS) vulnerabilities allow user-supplied data to be
+   incorrectly executed as code in a web browser. This can allow an attacker
+   to inject malicious scripts, which in the worst case can cause the loss
+   of private user data.
+
+   In order to keep your code safe from such attackers, you should follow the
+   best practices laid out in the guide
+   `Preventing Cross Site Scripting Vulnerabilities`_.
+
+Technology Selection
+~~~~~~~~~~~~~~~~~~~~
+
+1. **Server-side templating should use Django Templates**
+
+   Rationale: There are many template languages available for Django,
+   but the simplest option is to use the built-in Django template engine.
+   The Open edX codebase has a mixture of Django and Mako templates, but the
+   former are easier to reason about because they don't support arbitrary
+   code evaluation. See `Reasons for rejecting Mako`_ for a more detailed
+   explanation.
+
+   **Exception**: Mako templates should be used within edx-platform for
+   consistency with the existing code base. Unfortunately there has been too
+   much investment in Mako to consider replacing it at this point.
+
+2. **ECMAScript 2015 should be the supported version of JavaScript**
+
+   Rationale: ECMAScript is the standardized version of JavaScript,
+   and the most recently ratified version is ECMAScript 2015. It is
+   commonly referred to by ES2015, or by ES6 (its original name). Most
+   browsers now provide native support for ES2015, but it is generally
+   recommended to use `Babel`_ to transpile ES2015 code into ES5 which is
+   universally supported.
+
+3. **JavaScript libraries should be installed via npm**
+
+   Rationale: It is important that JavaScript libraries are kept
+   up-to-date, and one key component is to make it as simple as possible
+   to perform upgrades. Projects that use npm maintain all of their
+   library requirements in a single package.json file, which can be
+   easily changed as the versions change.
+
+4. **JavaScript code should follow the edX ESLint configuration**
+
+   Rationale: In order to standardize and enforce Open edX's JavaScript
+   coding style across multiple codebases, edX has published an ESLint
+   configuration that provides an enforceable specification. The
+   `edX ESLint Config`_ is made available as an npm package that can be
+   installed into any Open edX package.
+
+5. **Front-end interactivity should be added using Backbone**
```

Comment:
I am getting the impression that this is comparing apple to orange. React alone can't replace a MVC framework, w/o others such as Flux or Redux. Is the direction that we are moving towards Backbone + React or Redux + React and why the choice.

It's worth mentioning that React community is growing rapidly, that Open Source community that knows React is also the same. Maybe that is one of the arguments.